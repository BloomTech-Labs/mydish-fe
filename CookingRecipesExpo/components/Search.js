import React, {useState, useEffect} from "react";
import {View,TouchableOpacity, TextInput, Button, StyleSheet, Text, ScrollView, Image} from "react-native";
import axios from "axios";
import logo from '../assets/LogoGreen.png';
import RecipeList from './RecipeList.js'

const Search = () => {
    let [dish, setDish] = useState('')
    let [recipes, setRecipes] = useState([])
    let [recipeListRefresh, setRecipeListRefresh] = useState(false);


    useEffect(() =>{
        axios
        .get(
          `https://recipeshare-development.herokuapp.com/recipes?title=${dish}`
        )
        .then(res => {
            setRecipes([])          
            setRecipes(res.data);
        })
        .catch(err => console.log(err));
    },[dish,recipeListRefresh]);

    // const handleBlur = () => {
    //     console.log('handleBlur triggered in <Search>');
    //     setDish('');
    // }
    const focus = () => {
        console.log('focus on your search!');
        setDish('');
    }


    return(
        <View>
            <View style = {{flexDirection: 'row', justifyContent: 'left', textAlign: 'left', paddingBottom:"2%", marginTop: "2%"}}>
                <Image source={logo} style={{width: "8%", height: "85%", marginLeft: "2%"}}/> 
                <Text style={styles.title}>RecipeShare</Text>
            </View>
				 <TextInput
					style={styles.textInput}
					placeholder="What dish are you looking for?"
					placeholderTextColor="#D3D3D3"
					value={dish}
                    onChangeText={dish => setDish(dish)}
                    // clearTextOnFocus={true}
                    // onBlur={handleBlur}
                    onFocus={focus}
				/>

                <ScrollView>
                    {/* <TouchableOpacity onPress={refreshRecipeList}> */}
                        {recipes.length>=1  && <RecipeList recipes={recipes} setRecipes={setRecipes} /> }
                    {/* </TouchableOpacity> */}
                </ScrollView>
        </View>

    )
}

const styles = StyleSheet.create({
    title: {
        textAlign: "center",
        color: "#42C200",
        fontSize: 18,
        fontWeight: 'bold', 
        // paddingBottom:14,
        paddingTop: 5, 
        paddingLeft: 10
    },
    textInput: {
        //flex: 1,
        height: 40,
        //width: 300,
        fontSize: 18,
        margin: 7,
        fontWeight: 'bold',
        color: 'black',
        paddingLeft: 3,
        minHeight: '5%',
        borderRadius: 4,
        borderWidth: 1,
        borderColor: '#d6d7da'
    },
    button: {
        borderRadius: 4,
        borderWidth: 2,
        marginLeft: 100,
        marginRight: 100,
        borderColor: '#3BA405',
        backgroundColor: `#3BA405`
    }
})

export default Search;