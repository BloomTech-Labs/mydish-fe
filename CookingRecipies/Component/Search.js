import React, {useState} from "react";
import {View,View,TouchableOpacity, TextInput, Button} from "react-native";
import { FormInput} from 'react-native-elements'

const Search = () => {
    let [dish, setDish] = useState('')
    let [recipe, setRecipes] = useState([])
    
    let grabRecipes = dish => {
        
      };
  

    return(
        <View>
				<TextInput
					// style={styles.textInput}
					// multiline={true}
					placeholder="Dish?"
					placeholderTextColor="#abbabb"
					value={dish}
					onChangeText={dish => setDish(dish)}
				/>
                <TouchableOpacity>
                    <Button
                    onPress={grabRecipes}
                    title="Click me"
                    color="red"
                    accessibilityLabel="Search"
                    />
                </TouchableOpacity>
            <FormInput onChangeText={someFunction}/>
            {/* <RecipeList data={state.data}/> */}

        </View>

    )
}