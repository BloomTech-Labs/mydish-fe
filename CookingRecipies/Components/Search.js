import React, {useState, useEffect} from "react";
import {View,TouchableOpacity, TextInput, Button, StyleSheet, Text} from "react-native";
import axios from "axios";

import RecipeList from './RecipeList.js'

const Search = () => {
    let [dish, setDish] = useState('')
    let [recipe, setRecipes] = useState([])

    useEffect(() =>{
        axios
        .get(
          `https://restcountries.eu/rest/v2/all`
        )
        .then(res => {
          setRecipes([res.data]);
          console.log("hi", recipe)
        })
        .catch(err => console.log(err));

    },[]);
    
    let grabRecipes = e => {
        e.preventDefault();
        console.log(dish);
        if( dish.length>0){
            axios
            .get(
              `https://restcountries.eu/rest/v2/capital/${dish}`
            )
            .then(res => {
              setRecipes([]);
              setRecipes([res.data]);
              console.log(recipe)
              
            })
            .catch(err => console.log(err));
        };
    };


    return(
        <View>
            <Text >Search Bar</Text>
				 <TextInput
					style={styles.textInput}
					placeholder="What Dish are you looking for?"
					placeholderTextColor="#D3D3D3"
					value={dish}
					onChangeText={dish => setDish(dish)}
				/>
                 <TouchableOpacity style={styles.button}>
                    <Button  
                    color="white"    
                    onPress={grabRecipes}
                    title="Click me"
                    accessibilityLabel="Search"                   
                    />
                </TouchableOpacity>
            {/* <RecipeList />  */}


        </View>

    )
}

const styles = StyleSheet.create({
textInput: {
    flex: 1,
    height: 4000,
    fontSize: 18,
    margin: 7,
    fontWeight: 'bold',
    color: 'black',
    paddingLeft: 3,
    minHeight: '65%',
    borderRadius: 4,
    borderWidth: 1,
    borderColor: '#d6d7da'
},
button: {
    borderRadius: 4,
    borderWidth: 2,
    marginLeft: 100,
    marginRight: 100,
    borderColor: '#2089dc',
    backgroundColor: `#2089dc`
}
})

export default Search;