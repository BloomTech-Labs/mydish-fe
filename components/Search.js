import React from 'react';
import { TextInput } from 'react-native';
import styles from '../styles/search.styles';
//Analytics
import { Analytics, Event } from 'expo-analytics';
const analytics = new Analytics('UA-160806654-1');

const Search = (props) => {
  const { dish, setDish } = props;

  const searching = () => {
    analytics
      .event(new Event('Searchbar', 'User searching for recipes'))
      .then(() => console.log('Searchbar accessed'))
      .catch((e) => console.log(e.message));
  };

  return (
    <TextInput
      style={styles.textInput}
      placeholder="What dish are you looking for?"
      placeholderTextColor="#D3D3D3"
      value={dish}
      onChangeText={(text) => setDish(text)}
      onFocus={searching}
    />
  );
};

export default Search;
