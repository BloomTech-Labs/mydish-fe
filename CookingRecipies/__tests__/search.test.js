import React from 'react';
import Search from '../Components/Search';
import renderer from 'react-test-renderer';

test('search snapshot', () => {
    const searchSnap = renderer.create(<Search/>).toJSON();
    expect(searchSnap).toMatchSnapshot();
})



