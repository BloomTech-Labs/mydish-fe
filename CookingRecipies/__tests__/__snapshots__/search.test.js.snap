// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="What Dish are you looking for?"
    placeholderTextColor="#D3D3D3"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "black",
        "flex": 1,
        "fontSize": 18,
        "fontWeight": "bold",
        "height": 40,
        "margin": 7,
        "minHeight": "5%",
        "paddingLeft": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3BA405",
        "borderColor": "#3BA405",
        "borderRadius": 4,
        "borderWidth": 2,
        "marginLeft": 100,
        "marginRight": 100,
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityLabel="Search"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Click me
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;
